autolab/Autolab
# Button to clear cache (in dropdown)

## Description
Added a button to clear expired cache (cache cleanup).

### On desktop
<img width="1203" alt="Screenshot 2023-02-06 at 4 09 56 PM" src="https://user-images.githubusercontent.com/65066221/217088699-db2106c9-0985-4eb5-971b-52890fd60bd6.png">

### On mobile
<img width="519" alt="Screenshot 2023-02-06 at 4 12 01 PM" src="https://user-images.githubusercontent.com/65066221/217089114-5ed38ec4-1ddb-4944-839b-83aab76fea0f.png">

### After button is pressed
<img width="1199" alt="Screenshot 2023-02-06 at 4 12 19 PM" src="https://user-images.githubusercontent.com/65066221/217089174-6730cd1c-9a9f-40d0-acc3-a73168058ebd.png">

## Motivation and Context
Previously, an admin had to run the command <code>rake user:cleanup_cache</code> to clear expired cache #1634 . This button makes it easier to do so.

## How Has This Been Tested?

### Preliminary steps

- Enable caching in development by running <code>rails dev:cache</code>
- In development.rb, replace <code>config.cache_store = :memory_store</code> with <code>config.cache_store = :file_store, "tmp/cache/scores"</code>

### Test that rake task clears expired entries

- For testing purposes, change <code>:expires_in</code> and <code>race_condition_ttl</code> to <code>5.seconds</code> (in all 3 places)
- Load the gradesheet of an assessment (e.g. AutoPopulated / Homework0), observe that the path <code><AUTOLAB DIR>/tmp/cache/scores</code> is populated
- Wait a few seconds, click `Manage Autolab` followed by `Clear Cache`, observe that the (expired) entries are cleared

### Test that rake task does not clear unexpired entries

- Change <code>:expires_in</code> and <code>race_condition_ttl</code> back to their original values
- Load the gradesheet of an assessment (e.g. AutoPopulated / Homework0), observe that the path <code><AUTOLAB DIR>/tmp/cache/scores</code> is populated
- Wait a few seconds, click `Manage Autolab` followed by `Clear Cache`, observe that the unexpired entries are not cleared
- As a sanity check, run <code>rake user:clear_cache</code> and observe that the entries are cleared

## Types of changes

## Checklist:
